### PR TITLE
[remote/downloader] Migrate `Downloader` to take `Credentials`

### PR DESCRIPTION
**Pull request by Yannic**
Progress on https://github.com/bazelbuild/bazel/issues/15856

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/authandtls/staticcredentials/StaticCredentials.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  public StaticCredentials(Map<URI, Map<String, List<String>>> credentials) {
+    Preconditions.checkNotNull(credentials);
+
+    this.credentials = ImmutableMap.copyOf(credentials);
+  }
+
+  public Map<URI, Map<String, List<String>>> getMapForMigration() {
+    return credentials;
+  }
```
Comment (tjgq) [inline src/main/java/com/google/devtools/build/lib/authandtls/staticcredentials/StaticCredentials.java:23]: I couldn't find any callers for this method in this PR. Will it be used in a followup? If so, consider adding it in the same PR where it gets used, so we can later trace back why it was added.

**Review by tjgq — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/devtools/build/lib/authandtls/staticcredentials/StaticCredentials.java (hunk 1 above)
Comment (Yannic) [reply src/main/java/com/google/devtools/build/lib/authandtls/staticcredentials/StaticCredentials.java:23]: Eh, yeah. I used this when creating this PR. Removing

File: src/test/java/com/google/devtools/build/lib/bazel/repository/downloader/HttpDownloaderTest.java [hunk 2]
```diff
@@ -370,7 +371,7 @@ public void downloadOneUrl_ok() throws IOException, InterruptedException {
       httpDownloader.download(
           Collections.singletonList(
               new URL(String.format("http://localhost:%d/foo", server.getLocalPort()))),
-          Collections.emptyMap(),
+          new StaticCredentials(ImmutableMap.of()),
```
[the remark this answers is not in the document]
Comment (Yannic) [reply src/test/java/com/google/devtools/build/lib/bazel/repository/downloader/HttpDownloaderTest.java:374]: done

**Comment by Yannic** [comment]
PTAL

**Review by tjgq — APPROVED** [review]
Importing.

I think I'd prefer to put StaticCredentials.java in `authandtls` instead of creating another subpackage (unless you give me a strong reason not to). Thoughts?

**Pull request closed without merging**